### ISSUE TITLE
Avoid users to create empty datasources

### ISSUE THREAD

**Issue by rohan-arthur** [appsmithorg/appsmith#32143]
https://www.notion.so/Avoid-users-to-create-empty-datasources-575f98fdaee84702beea92262d15a545

**Comment by rishabhrathod01** [comment]
@rohan-arthur this doesn’t look like a small effort task to the team. Could we re-review this ballpark estimation?

**Comment by rohan-arthur** [comment]
@rishabhrathod01 of course, please change the estimation at any time, with any useful info that can help us do better next time!
